projects/tiempoCarlaSergio/src/components/Buscador.vue

```diff
@@ -27,13 +27,18 @@ export default {
     }
   },
   mounted (){
-    
+    if(this.$route.query.ubicacion){
+      this.search(this.$route.query.ubicacion);
+    }
   },
   methods: {
-    search: function() {
+    changeUrl(){
+      this.$router.push({ name: 'detalles', query: { ubicacion: this.param } }).catch(err => { });
```

Comment:
Si solo vais moditificar la 'query' de la ruta actual es mejor usar 'replace'.

this.$router.replace({ name: 'detalles', query: { ubicacion: this.param } })

De esta manera no crea una nueva entrada en el histórico de navegación. De lo contrario al darle a la flecha de 'atrás', en el navegador, cargará de nuevo la página en vez de ir a la página anterior.

https://router.vuejs.org/guide/essentials/navigation.html#router-replace-location-oncomplete-onabort